V9: segfault scanning k8s

**Describe the bug**
```
cnquery scan k8s --log-level trace
DBG using provider k8s with connector k8s
DBG no need to update provider last-refresh=40m34.54718321s provider=k8s
DBG running provider plugin path=/Users/tsmith/.config/mondoo/providers/k8s/k8s
→ loaded configuration from /Users/tsmith/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
DBG adding 83 discovered asset(s)
DBG running provider plugin path=/Users/tsmith/.config/mondoo/providers/os/os
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x98 pc=0x10146f05c]

goroutine 1 [running]:
go.mondoo.com/cnquery/explorer/scan.(*LocalScanner).distributeJob(0x140008db230, 0x140008b0fc0, {0x1019d14f8, 0x140008db260}, 0x14000752c40)
	/Users/tsmith/dev/cnquery/explorer/scan/local_scanner.go:241 +0x7ac
go.mondoo.com/cnquery/explorer/scan.(*LocalScanner).Run(0x1019d14c0?, {0x1019d14f8?, 0x140008db260?}, 0x10237e7c0?)
	/Users/tsmith/dev/cnquery/explorer/scan/local_scanner.go:92 +0x50
go.mondoo.com/cnquery/apps/cnquery/cmd.RunScan(0x140008df480)
	/Users/tsmith/dev/cnquery/apps/cnquery/cmd/scan.go:287 +0x364
go.mondoo.com/cnquery/apps/cnquery/cmd.glob..func21(0x0?, 0x140002f9b00?, 0x23?)
	/Users/tsmith/dev/cnquery/apps/cnquery/cmd/scan.go:119 +0x94
go.mondoo.com/cnquery/cli/providers.setConnector.func2(0x14000347c00?, {0x14000385ce0?, 0x0, 0x2})
	/Users/tsmith/dev/cnquery/cli/providers/providers.go:336 +0xc8c
github.com/spf13/cobra.(*Command).execute(0x1400039b500, {0x14000385cc0, 0x2, 0x2})
	/Users/tsmith/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x640
github.com/spf13/cobra.(*Command).ExecuteC(0x1023334c0)
	/Users/tsmith/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/tsmith/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
go.mondoo.com/cnquery/apps/cnquery/cmd.Execute()
	/Users/tsmith/dev/cnquery/apps/cnquery/cmd/root.go:71 +0x15c
main.main()
	/Users/tsmith/dev/cnquery/apps/cnquery/cnquery.go:9 +0x1c
```

--- a/explorer/scan/local_scanner.go
+++ b/explorer/scan/local_scanner.go
@@ -235,12 +235,7 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 
 	justAssets := []*inventory.Asset{}
 	for _, asset := range assets {
-		if asset.asset.GetPlatform() != nil {
-			asset.asset.KindString = asset.asset.GetPlatform().Kind
-		} else {
-			asset.asset = asset.runtime.Provider.Connection.Asset
-			asset.asset.KindString = asset.runtime.Provider.Connection.Asset.Platform.Kind
-		}
+		asset.asset.KindString = asset.asset.GetPlatform().Kind
 		justAssets = append(justAssets, asset.asset)
 	}
 

--- a/explorer/scan/local_scanner.go
+++ b/explorer/scan/local_scanner.go
@@ -188,15 +188,11 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 			log.Error().Err(err).Msg("unable to connect to asset")
 			continue
 		}
-		inventorySpec := runtime.Provider.Connection
-		if inventorySpec.Inventory != nil &&
-			inventorySpec.Inventory.Spec != nil &&
-			inventorySpec.Inventory.Spec.Assets != nil {
-			log.Debug().Msgf("adding %d discovered asset(s)", len(runtime.Provider.Connection.Inventory.Spec.Assets))
-			assetCandidates = append(assetCandidates, inventorySpec.Inventory.Spec.Assets...)
-		} else {
-			assetCandidates = append(assetCandidates, runtime.Provider.Connection.Asset)
+		processedAssets, err := providers.ProcessAssetCandidates(runtime, runtime.Provider.Connection, upstream, "")
+		if err != nil {
+			return nil, false, err
 		}
+		assetCandidates = append(assetCandidates, processedAssets...)
 		// TODO: we want to keep better track of errors, since there may be
 		// multiple assets coming in. It's annoying to abort the scan if we get one
 		// error at this stage.
@@ -239,7 +235,12 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 
 	justAssets := []*inventory.Asset{}
 	for _, asset := range assets {
-		asset.asset.KindString = asset.asset.GetPlatform().Kind
+		if asset.asset.GetPlatform() != nil {
+			asset.asset.KindString = asset.asset.GetPlatform().Kind
+		} else {
+			asset.asset = asset.runtime.Provider.Connection.Asset
+			asset.asset.KindString = asset.runtime.Provider.Connection.Asset.Platform.Kind
+		}
 		justAssets = append(justAssets, asset.asset)
 	}
 

--- a/providers/aws/provider/provider.go
+++ b/providers/aws/provider/provider.go
@@ -13,16 +13,12 @@ import (
 	"go.mondoo.com/cnquery/providers-sdk/v1/upstream"
 	"go.mondoo.com/cnquery/providers/aws/connection"
 	"go.mondoo.com/cnquery/providers/aws/resources"
-	osconnection "go.mondoo.com/cnquery/providers/os/connection"
 	"go.mondoo.com/cnquery/providers/os/connection/shared"
-	"go.mondoo.com/cnquery/providers/os/detector"
 )
 
 const (
-	defaultConnection           uint32 = 1
-	DefaultConnectionType              = "aws"
-	SshConnectionType                  = "ssh"
-	RegistryImageConnectionType        = "registry-image"
+	defaultConnection     uint32 = 1
+	DefaultConnectionType        = "aws"
 )
 
 type Service struct {
@@ -134,20 +130,9 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 	var err error
 
 	switch conf.Type {
-	case SshConnectionType:
-		s.lastConnectionID++
-		conn, err = osconnection.NewSshConnection(s.lastConnectionID, conf, asset)
-		if pf, ok := detector.DetectOS(conn); ok {
-			conn.Asset().Platform = pf
-		}
-
-	case RegistryImageConnectionType:
-		s.lastConnectionID++
-		conn, err = osconnection.NewContainerRegistryImage(s.lastConnectionID, conf, asset)
 	default:
 		s.lastConnectionID++
 		conn, err = connection.NewAwsConnection(s.lastConnectionID, asset, conf)
-
 	}
 	if err != nil {
 		return nil, err

